### PR TITLE
Fixes issue where expired and expiring notice displayed incorrectly

### PR DESCRIPTION
**Pull request by gedex**
Consider connected (to the current site) product when checking expired and expiring notice.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

* Use `$subscription['active']` to filter out subscriptions that are not connected to the current site when using `WC_Helper::get_subscription_list_data()`. This is for screens in wc settings and in-apps.
* Same approach on above for plugin list screen, but given it uses `WC_Helper::get_subscriptions()` and there's no `$subscription['active']`, use `$subscription['connections']`.

#### Expired subscription

**expectations**

* When there's an expired subscription that's installed but not connected / activated on the site, notice shouldn't be shown.
* When there's an expired subscription that's installed and connected / activated on the site, notice should be shown.
* Have two subscriptions of the same product, one is expired and connected the site, notice should be shown.
* Have two subscriptions of the same product, one is not expired and connected the site, notice shouldn't be shown.

**in-apps screen**

image

**wc setting screen**

image

**plugin screen**

image

#### Expiring subscriptions

**expectations**

* When there's an expiring subscription that's installed but not connected / activated on the site, notice shouldn't be shown.
* When there's an expiring subscription that's installed and connected / activated on the site, notice should be shown.
* Have two subscriptions of the same product, one is expiring and connected the site, notice should be shown.
* Have two subscriptions of the same product, one is not expiring and connected the site, notice shouldn't be shown.

**in-apps screen**

image

**wc setting screen**

image

**plugin screen**

image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/includes/admin/helper/class-wc-helper.php [hunk 1]
```diff
@@ -1297,6 +1297,31 @@ public static function has_product_subscription( $product_id ) {
...
+	 *
+	 * @return array
+	 */
+	public static function get_installed_subscriptions() {
+		static $installed_subscriptions = null;
+
+		// Cache installed_subscriptions in the current request.
+		if ( is_null( $installed_subscriptions ) ) {
+			$auth    = WC_Helper_Options::get( 'auth' );
+			$site_id = isset( $auth['site_id'] ) ? absint( $auth['site_id'] ) : 0;
```
Comment (thilinah) [inline plugins/woocommerce/includes/admin/helper/class-wc-helper.php:1312]: Maybe we can return early here, after setting `$installed_subscriptions = []`, when `$auth['site_id']` isn't set.

**Review by thilinah — APPROVED** [review]
Thanks @gedex. Tested and works fine. Left a minor comment.

File: plugins/woocommerce/includes/admin/helper/class-wc-helper.php (hunk 1 above)
Comment (gedex) [reply plugins/woocommerce/includes/admin/helper/class-wc-helper.php:1312]: Good point. I'll update it.

Comment (gedex) [reply plugins/woocommerce/includes/admin/helper/class-wc-helper.php:1312]: Updated in https://github.com/woocommerce/woocommerce/pull/49717/commits/132c8dff227d21975815bb5c42795b543df17181.

**Pull request merged**